src/core/manager.c

```diff
@@ -3306,6 +3315,80 @@ static int manager_deserialize_units(Manager *m, FILE *f, FDSet *fds) {
         return 0;
 }
 
+usec_t manager_get_runtime_watchdog(Manager *m) {
+        assert(m);
+
+        if (MANAGER_IS_USER(m))
+                return USEC_INFINITY;
+
+        if (timestamp_is_set(m->runtime_watchdog_overridden))
+                return m->runtime_watchdog_overridden;
+
+        return m->runtime_watchdog;
+}
+
+usec_t manager_get_reboot_watchdog(Manager *m) {
+        assert(m);
+
+        if (timestamp_is_set(m->reboot_watchdog_overridden))
+                return m->reboot_watchdog_overridden;
+
+        return m->reboot_watchdog;
+}
+
+usec_t manager_get_kexec_watchdog(Manager *m) {
+        assert(m);
+
+        if (timestamp_is_set(m->kexec_watchdog_overridden))
+                return m->kexec_watchdog_overridden;
+
+        return m->kexec_watchdog;
+}
```

Comment:
I think this would all become a lot prettier and shorter if we'd start storing the three types of watchdog settings in an array, indexed by an enum. We could then always just specify via the enum which of the three watchdog settings we mean...

i.e. something like this:

```c
typedef enum WatchdogType {
        WATCHDOG_RUNTIME,
        WACTHDOG_REBOOT,
        WATCHDOG_KEXEC,
        _WATCHDOG_TYPE_MAX,
} WatchdogType;
```

And then in the Manager object:

```
…
        usec_t watchdog_overriden[_WATCHDOG_TYPE_MAX];
…
```

And these functions become:

```
usec_t manager_get_watchdog_usec(Manager *m, WatchdogType t) {
…
}
```

you get the idea...